Yosemite/YosemiteTests/Stores/CustomerStoreTests.swift

```diff
@@ -0,0 +1,167 @@
+    ///
+    private var viewStorage: StorageType {
+        return storageManager.viewStorage
+    }
+
+    /// Testing SiteID
+    ///
+    private let sampleSiteID: Int64 = 1234
+
+    // MARK: - Overridden Methods
+
+    override func setUp() {
+        super.setUp()
+        dispatcher = Dispatcher()
+        storageManager = MockStorageManager()
+        network = MockNetwork(useResponseQueue: true)
+    }
+
+    override func tearDown() {
+        dispatcher = nil
+        storageManager = nil
+        network = nil
+
+        super.tearDown()
+    }
+
+    // MARK: - CustomerAction.synchronizeAllCustomers
+
+    func test_synchronizeAllCustomers_persists_all_pages_of_customers_on_success() throws {
+        // Given
+        let store = CustomerStore(dispatcher: dispatcher, storageManager: storageManager, network: network)
+        network.simulateResponse(requestUrlSuffix: "customers", filename: "customers-all")
+        network.simulateResponse(requestUrlSuffix: "customers", filename: "customers-extra")
+        network.simulateResponse(requestUrlSuffix: "customers", filename: "customers-empty")
+        XCTAssertEqual(viewStorage.countObjects(ofType: Storage.Customer.self), 0)
+
+        // When
+        let result: Result<Void, Error> = try waitFor { promise in
+            let action = CustomerAction.synchronizeAllCustomers(siteID: self.sampleSiteID) { result in
+                promise(result)
+            }
+            store.onAction(action)
+        }
+
+        // Then
+        XCTAssertTrue(result.isSuccess)
+        XCTAssertEqual(viewStorage.countObjects(ofType: Storage.Customer.self), 3)
+    }
+
+    func test_synchronizeAllCustomers_persists_customer_details_on_success() throws {
+        // Given
+        let store = CustomerStore(dispatcher: dispatcher, storageManager: storageManager, network: network)
+        network.simulateResponse(requestUrlSuffix: "customers", filename: "customers-all")
+        network.simulateResponse(requestUrlSuffix: "customers", filename: "customers-empty")
+        XCTAssertEqual(viewStorage.countObjects(ofType: Storage.Customer.self), 0)
+        let remoteCustomer = try sampleCustomer()
+
+        // When
+        let result: Result<Void, Error> = try waitFor { promise in
+            let action = CustomerAction.synchronizeAllCustomers(siteID: self.sampleSiteID) { result in
+                promise(result)
+            }
+            store.onAction(action)
+
+        }
+
+        // Then
+        XCTAssertTrue(result.isSuccess)
+        let storedCustomer = self.viewStorage.loadCustomer(siteID: remoteCustomer.siteID, userID: remoteCustomer.userID)
+        let readOnlyCustomer = storedCustomer?.toReadOnly()
+        XCTAssertNotNil(storedCustomer)
+        XCTAssertNotNil(readOnlyCustomer)
+        XCTAssertEqual(readOnlyCustomer, remoteCustomer)
+    }
+
+    func test_synchronizeAllCustomers_returns_error_on_failure() throws {
```

Comment:
nit: It's a good thing here to check that also the number of stored Customer is still zero 👍

Reply:
Added in 82349ff.

Reply:
Added in 82349ff.